### PR TITLE
feat: REPLAT-10475 draft article query and provider created

### PR DESCRIPTION
**Pull request by infreezer**
[Story.](https://nidigitalsolutions.jira.com/browse/REPLAT-10475)

Draft article query created.
Draft article provider created.
isPreview boolean consumed in ssr.

image

### CODE REVIEW

File: packages/ssr/src/component/article.js [hunk 1]
```diff
@@ -37,14 +39,16 @@ module.exports = (client, analyticsStream, data, helmetContext) => {
...
-        ArticleProvider,
+        isPreview ? DraftArticleProvider : ArticleProvider,
         {
           analyticsStream,
           debounceTimeMs,
           id: articleId
         },
-        ({ article, isLoading, error, refetch }) =>
-          React.createElement(
+        ({ article, draftArticle, isLoading, error, refetch }) => {
```
Comment (aronshamash) [inline packages/ssr/src/component/article.js]: hmm not sure I understand why we need this ?

File: packages/ssr/src/component/article.js [hunk 2]
```diff
@@ -37,14 +39,16 @@ module.exports = (client, analyticsStream, data, helmetContext) => {
...
+          const articleData = isPreview ? draftArticle : article;
```
Comment (apostolnikov) [inline packages/ssr/src/component/article.js]: I don't think that it's good idea to always have the 2 fields from the two providers as destruct parameters, as one of them will be missing all the times. Could we use this articleData logic at upper level so we can have it inside the destructuring as well?

File: packages/ssr/src/component/article.js (hunk 1 above)
Comment (infreezer) [reply packages/ssr/src/component/article.js]: When the draft article provider is used the data is returned in draftArticle variable

File: packages/ssr/src/component/article.js (hunk 2 above)
Comment (aronshamash) [reply packages/ssr/src/component/article.js]: yes my thoughts too.

Comment (infreezer) [reply packages/ssr/src/component/article.js]: Comments addressed. Please review.

**Pull request merged**